amid/utils.py

```diff
@@ -0,0 +1,53 @@
+import contextlib
+import zipfile
+from gzip import GzipFile
+from pathlib import Path
+
+import nibabel
+
+
+@contextlib.contextmanager
+def unpack(root: str, relative: str, archive_root_name: str = None):
+    """Provides the absolute path to the file in both scenarios: inside archive or inside folder.
+
+    Parameters
+    ----------
+    root : str, Path
+        Absolute path to the downloaded archive or the unpacked archive root.
+    relative : str, Path
+        Relative file path inside the archive. Archive's root folder sholud be ommited.
+    archive_root_name : str, Path, optional
+        If `root` is a archive, it's root folder name shold be given.
+
+    Returns
+    -------
+    unpacked : Path
+        Absolute file path to be opened.
+    is_unpacked : {True, False}
+        Reached file state. `True` if the file is located inside archive, `False` otherwise.
+    """
+    unpacked = Path(root) / relative
+
+    if unpacked.exists():
+        yield unpacked, True
+    elif root.name.endswith('.zip'):
```

Comment:
maybe pass the extension as an optional argument? this way we won't rely on the correctness of the file name and just use it as a fallback